src/mesh/config/MeshConfiguration.cpp

```diff
@@ -52,30 +58,25 @@ MeshConfiguration::MeshConfiguration(
   parent.addSubtag(tag);
 }
 
-void MeshConfiguration::setDimensions(
-    int dimensions)
-{
-  PRECICE_TRACE(dimensions);
-  PRECICE_ASSERT((dimensions == 2) || (dimensions == 3), dimensions);
-  _dimensions = dimensions;
-}
-
 void MeshConfiguration::xmlTagCallback(
     const xml::ConfigurationContext &context,
     xml::XMLTag &                    tag)
 {
   PRECICE_TRACE(tag.getName());
   if (tag.getName() == TAG) {
-    PRECICE_ASSERT(_dimensions != 0);
-    std::string name = tag.getStringAttributeValue(ATTR_NAME);
+    std::string name       = tag.getStringAttributeValue(ATTR_NAME);
+    int         dimensions = tag.getIntAttributeValue(ATTR_DIMENSIONS);
+    _meshDimensionsMap.insert(std::pair<std::string, int>(name, dimensions));
+    PRECICE_ASSERT(dimensions != 0);
     PRECICE_ASSERT(_meshIdManager);
-    _meshes.push_back(std::make_shared<Mesh>(name, _dimensions, _meshIdManager->getFreeID()));
+    _meshes.push_back(std::make_shared<Mesh>(name, dimensions, _meshIdManager->getFreeID()));
```

Comment:
Since we now have each mesh having potentially different dimensions, we replace the global `int dimensions` with a map from mesh name to dimensions.

These dimensions are not only a local variable.